dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/MessageFormatter.java

```diff
@@ -108,7 +108,7 @@ private static String formatArgument(TrackerIdSchemeParams idSchemes, Object arg
     if (Instant.class.isAssignableFrom(type))
       return DateUtils.toIso8601NoTz(DateUtils.fromInstant((Instant) argument));
     if (Enrollment.class.isAssignableFrom(type)) return ((Enrollment) argument).getEnrollment();
-    if (Event.class.isAssignableFrom(type)) return ((Event) argument).getEvent();
+    if (Event.class.isAssignableFrom(type)) return ((Event) argument).getEvent().getValue();
```

Comment:
Can you please double-check that a UID argument is correctly formatted by the MessageFormatter.

Reply:
Added `formatArgumentsShouldTurnEventIntoArguments` in `MessageFormatterTest`